### ISSUE TITLE
Add DXR Support

### ISSUE THREAD

**Issue by Mayhem50** [sharpdx/SharpDX#1029]
Microsoft has released several days ago his DirectX12 SDK for DXR. It could be interesting to add this to SharpDX.

[https://www.youtube.com/watch?v=yaK1e59-oAA](url)
[https://onedrive.live.com/?authkey=%21AOmqJ4vut9zN3Ss&cid=95FC1A2974BC6D6B&id=95FC1A2974BC6D6B%21108&parId=root&action=locate](url)

**Comment by nerogann** [comment]
Any update for DXR?

**Comment by SouldomainTM** [comment]
Yeah, since Nvidia hit us now with hardware accelerated ray tracing with the Geforce RTX series. It's time that we got coding at some point.

**Comment by Ziriax** [comment]
We are developing a [GLTF 2.0](https://www.khronos.org/gltf/) compliant rendering engine using SharpDX and DX11.3, and we will most likely need DXR/RTX support at some point. We also created [Maya2glTF](https://github.com/WonderMediaProductions/Maya2glTF) to get full control over the export process and advanced features. 

But it can be a couple of months before we start coding this. 

If anyone starts working on this, please mention it here, so we can collaborate?

**Comment by SouldomainTM** [comment]
@Ziriax , what kind of rendering engine is it: Proprietary, payed, open source?

I'm making an F# game engine. If I had a fitting renderer, I may use that instead of my own. And in your case, I could start earlier to work on the bindings. Seems for some reason DXR ain't for any DX11 feature levels anymore. I would have to do some DX12 first if I went to make my own renderer instead.

**Comment by ChuckNovice** [comment]
We definitely need DXR in SharpDX. I got a DX12 renderer that is almost ready so I could contribute at some point.

**Comment by Ziriax** [comment]
@SouldomainTM 

We're not yet sure if we're going to open source it. It is not a renderer for games, but for real-time character animation and visualization. So it is not optimized for games, it uses 8-layers of order-independent-transparancy using ROVs, GLTF PBR shading, compute-shaders for doing skinning/morphing/deformation, moment shadow mapping, ... Too heavy for a typical game engine I guess.  

We started with DX11, mainly to learn the API, but we will very likely have to switch to DX12, unless Microsoft brings DXR to DX11 too...

The only reason we're doing this, is because Epic didn't want to give us a license for Unreal Engine 4...
